### DISCUSSION TITLE
"develop" Branch for Beta Release?

### DISCUSSION THREAD

**Discussion by JimmyKenMerchant** [LMMS/lmms#6281] [Project Management]
I humbly believe that Beta or Release will be published. I tested 1.3.0-a and confirmed it has brand new GUI designs and futures. How about adding "develop" and accept pull requests on the branch, and making "master" for Beta or Release?

**Comment by ghost** [comment]
Yeah..any sign towards 1.3 would be welcome.

**Comment by Veratil** [comment]
Eventually, if I remember the all the chats correctly, the master branch will be a rolling release and we'll release tagged versions as it goes.

**Comment by ghost** [comment]
I'm ok with that, but the last rolling/alpha/appimage  release was in 2021-04-26

**Comment by Spekular** [comment]
Hence "eventually". Adding a develop branch would not solve the issue of more frequent releases, and the proposal is just an inverse of the workflow we've used for 1.2 (create a stable branch and allow master to develop further). If you want signs of 1.3 I suggest checking the repository's commit history, and in particular the issue for the ongoing reorganization that we intend to finish before 1.3.

**Comment by JimmyKenMerchant** [comment]
Audacity uses master branch as develop, [using beta release (Audacity's releases page on Github)](https://github.com/audacity/audacity/releases). Seems the tail commit on each on master. You can notify the time for bug fix by tagging beta, and stopping to merge pull requests of new futures.

**Comment by ItsCubeTime** [comment]
Came to say the same thing. Its been almost 2 years since the last release for a software that gets plenty of weekly commits.

Unless the developers are rewriting major parts of the software from scratch - resulting in lots of bugs that are yet to be smashed, a new release anytime soon would be more than feasible lol

**Reply by Veratil** [reply]
We're finishing up two final steps in a reorganization process which has unfortunately taken a long time. After that we can make a new alpha build and then start focusing on bigger changes we want to do.

**Reply by Veratil** [reply]
That being said, there are ways to get newer builds from the pull requests themselves since our CI does full builds. ;)

**Reply by Veratil** [reply]
Here's the latest merged PR builds: https://github.com/LMMS/lmms/pull/6320#issuecomment-1054694313

**Reply by Veratil** [reply]
disclaimer: don't use those builds for anything you are working on with the 1.2.2 version without making backups

**Comment by tresf** [comment]
I know I'm a year late on this discussion, but a developer got 99% there in releasing nightlies here: https://github.com/LMMS/lmms.io/pull/351.

**Reply by tresf** [reply]
Another year later, nightlies are working.  We even have pull-request installers available now via `https://lmms.io/download/pull-request/<ID>` so you can install and test a pending feature without compiling it.

**Comment by charbelnicolas** [comment]
How come Linux's nightlies require wine?

**Reply by tresf** [reply]
They shouldn't "require" wine, but fail silently if wine is missing.  Is that not the case?

**Reply by charbelnicolas** [reply]
I have not installed it since it requires to download wine, so I don't really know. I'd rather not have to download wine needlessly. I'll just wait another decade for the full stable release I guess.

**Reply by tresf** [reply]
Where are you getting your nighties from?  Our nightlies are AppImages, which shouldn't require downloading of anything.

Wine is used for VST2 plugins, but is optional. Waiting for a stable release won't magically fix an issue with the installer.  Furthermore, snarky remarks about the release cycle just make the few devs we have more salty.  It's not welcome and it won't be tolerated.

**Reply by charbelnicolas** [reply]
I'm using lmms-git from the Arch User Repository which requires wine. I didn't know you had AppImages.

> Furthermore, snarky remarks about the release cycle just make the few devs we have more salty. It's not welcome and it won't be tolerated.

Sorry about that :)

**Reply by tresf** [reply]
> I'm using lmms-git from the Arch User Repository which requires wine. I didn't know you had AppImages.

Gotcha.  Yeah, you'll have to take that up with the Arch maintainer.  We don't provide any packages for Arch directly, so forcing wine as a dependency is a decision of that AUR maintainer.  You can probably fork that AUR and just comment out the wine stuff.

Historically on platforms like Debian, they would call the packages `lmms` and `lmms-full` or something like that to allow users to side-step the wine requirement, but these packages are always maintained by contributors of those distros so we don't have control over those decisions.

There's also a ~~snap~~ [flatpak package available](https://flathub.org/apps/io.lmms.LMMS).  It's not officially maintained by us either although that maintainer does file bugs here occasionally so it's pretty close to being official.  I think the flatpak version will install Wine also (the snap version of wine?), but I haven't tried it myself.

**Edit:** I don't think there's a nightly flatpak, so that probably isn't useful in the context of this conversation.

**Reply by charbelnicolas** [reply]
Alright, thank you, I'll see what I can do!

**Comment by Rossmaxx** [comment]
Update: nightlies are a thing now.
